src/ray/gcs/gcs_server/test/gcs_server_rpc_test.cc

```diff
@@ -1,88 +1,189 @@
+  }
 
+  rpc::ActorTableData genActorTableData(const JobID &job_id) {
+    rpc::ActorTableData actor_table_data;
+    TaskID task_id = TaskID::ForDriverTask(job_id);
+    ActorID actor_id = ActorID::Of(job_id, task_id, 0);
+    actor_table_data.set_actor_id(actor_id.Binary());
+    actor_table_data.set_job_id(job_id.Binary());
+    actor_table_data.set_state(
+        rpc::ActorTableData_ActorState::ActorTableData_ActorState_ALIVE);
+    actor_table_data.set_max_reconstructions(2);
+    actor_table_data.set_remaining_reconstructions(2);
+    return actor_table_data;
+  }
+
+ protected:
+  // gcs server
+  std::unique_ptr<std::thread> thread_;
+  std::unique_ptr<rpc::GrpcServer> server_;
+  std::unique_ptr<rpc::JobInfoGrpcService> job_info_service_;
+  std::unique_ptr<rpc::DefaultJobInfoHandler> job_info_handler_;
+  std::unique_ptr<rpc::ActorInfoGrpcService> actor_info_service_;
+  std::unique_ptr<rpc::DefaultActorInfoHandler> actor_info_handler_;
+  gcs::GcsClientOptions options_;
+  std::shared_ptr<gcs::RedisGcsClient> redis_gcs_client_;
+  boost::asio::io_service io_service_;
+
+  // gcs client
+  std::unique_ptr<rpc::GcsRpcClient> client_;
+  std::unique_ptr<rpc::ClientCallManager> client_call_manager_;
+};
+
+TEST_F(GcsServerTest, TestActorInfo) {
+  // create actor_table_data
+  JobID job_id = JobID::FromInt(1);
+  rpc::ActorTableData actor_table_data = genActorTableData(job_id);
+
+  // Async register actor
+  rpc::ActorAsyncRegisterRequest asyncRegisterRequest;
+  asyncRegisterRequest.mutable_actor_table_data()->CopyFrom(actor_table_data);
+  TestAsyncRegister(asyncRegisterRequest);
+
+  // Async update actor state and get
+  rpc::ActorAsyncUpdateRequest asyncUpdateRequest;
+  asyncUpdateRequest.set_actor_id(actor_table_data.actor_id());
+  asyncUpdateRequest.mutable_actor_table_data()->CopyFrom(actor_table_data);
+  TestAsyncUpdateAndGet(asyncUpdateRequest);
+}
+
+TEST_F(GcsServerTest, TestJobInfo) {
+  // create job_table_data
+  JobID job_id = JobID::FromInt(1);
+  rpc::JobTableData job_table_data = genJobTableData(job_id);
   rpc::AddJobRequest add_job_request;
-  std::promise<rpc::AddJobReply> add_job_reply_promise;
-  auto add_job_reply_future = add_job_reply_promise.get_future();
-  client.AddJob(add_job_request, [&add_job_reply_promise](const Status &status,
-                                                          const rpc::AddJobReply &reply) {
-    if (status.ok()) {
-      add_job_reply_promise.set_value(reply);
-    }
-  });
-  auto future_status = add_job_reply_future.wait_for(std::chrono::milliseconds(200));
-  ASSERT_EQ(future_status, std::future_status::ready);
+  add_job_request.mutable_data()->CopyFrom(job_table_data);
+  uint64_t count = job_info_handler_->GetAddJobCount();
+  client_->AddJob(
+      add_job_request, [](const Status &status, const rpc::AddJobReply &reply) {
+        RAY_LOG(INFO) << "Received GcsServer AddJob response." << reply.DebugString();
+        ASSERT_TRUE(reply.success());
```

Comment:
What about if the async callback never be triggered? In that case, this unit test will be mistaken for good.

Reply:
fixed